// @ariakit/react-core/cjs/__chunks/3MQN464L.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";



var _32EK4BGYcjs = require('./32EK4BGY.cjs');


var _M5IWBORBcjs = require('./M5IWBORB.cjs');

// src/composite/composite-context.tsx
var _react = require('react');
var ctx = _M5IWBORBcjs.createStoreContext.call(void 0, 
  [_32EK4BGYcjs.CollectionContextProvider],
  [_32EK4BGYcjs.CollectionScopedContextProvider]
);
var useCompositeContext = ctx.useContext;
var useCompositeScopedContext = ctx.useScopedContext;
var useCompositeProviderContext = ctx.useProviderContext;
var CompositeContextProvider = ctx.ContextProvider;
var CompositeScopedContextProvider = ctx.ScopedContextProvider;
var CompositeItemContext = _react.createContext.call(void 0, 
  void 0
);
var CompositeRowContext = _react.createContext.call(void 0, 
  void 0
);









exports.useCompositeContext = useCompositeContext; exports.useCompositeScopedContext = useCompositeScopedContext; exports.useCompositeProviderContext = useCompositeProviderContext; exports.CompositeContextProvider = CompositeContextProvider; exports.CompositeScopedContextProvider = CompositeScopedContextProvider; exports.CompositeItemContext = CompositeItemContext; exports.CompositeRowContext = CompositeRowContext;
