// @ariakit/react-core/cjs/__chunks/5KP7SROC.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";


var _MUQXBDURcjs = require('./MUQXBDUR.cjs');


var _VRPZ4HD5cjs = require('./VRPZ4HD5.cjs');


var _6UCNERDTcjs = require('./6UCNERDT.cjs');


var _C55AGIGAcjs = require('./C55AGIGA.cjs');


var _WT6AUJWVcjs = require('./WT6AUJWV.cjs');



var _3IVNQNIVcjs = require('./3IVNQNIV.cjs');

// src/dialog/dialog-backdrop.tsx
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
function DialogBackdrop({
  store,
  backdrop,
  alwaysVisible,
  hidden
}) {
  const ref = _react.useRef.call(void 0, null);
  const disclosure = _C55AGIGAcjs.useDisclosureStore.call(void 0, { disclosure: store });
  const contentElement = store.useState("contentElement");
  _WT6AUJWVcjs.useSafeLayoutEffect.call(void 0, () => {
    const backdrop2 = ref.current;
    const dialog = contentElement;
    if (!backdrop2) return;
    if (!dialog) return;
    backdrop2.style.zIndex = getComputedStyle(dialog).zIndex;
  }, [contentElement]);
  _WT6AUJWVcjs.useSafeLayoutEffect.call(void 0, () => {
    const id = contentElement == null ? void 0 : contentElement.id;
    if (!id) return;
    const backdrop2 = ref.current;
    if (!backdrop2) return;
    return _MUQXBDURcjs.markAncestor.call(void 0, backdrop2, id);
  }, [contentElement]);
  const props = _VRPZ4HD5cjs.useDisclosureContent.call(void 0, {
    ref,
    store: disclosure,
    role: "presentation",
    "data-backdrop": (contentElement == null ? void 0 : contentElement.id) || "",
    alwaysVisible,
    hidden: hidden != null ? hidden : void 0,
    style: {
      position: "fixed",
      top: 0,
      right: 0,
      bottom: 0,
      left: 0
    }
  });
  if (!backdrop) return null;
  if (_react.isValidElement.call(void 0, backdrop)) {
    return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _6UCNERDTcjs.Role, _3IVNQNIVcjs.__spreadProps.call(void 0, _3IVNQNIVcjs.__spreadValues.call(void 0, {}, props), { render: backdrop }));
  }
  const Component = typeof backdrop !== "boolean" ? backdrop : "div";
  return /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _6UCNERDTcjs.Role, _3IVNQNIVcjs.__spreadProps.call(void 0, _3IVNQNIVcjs.__spreadValues.call(void 0, {}, props), { render: /* @__PURE__ */ _jsxruntime.jsx.call(void 0, Component, {}) }));
}



exports.DialogBackdrop = DialogBackdrop;
