// @ariakit/react-core/cjs/__chunks/AVQVVBHC.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";


var _MA4SJD6Ecjs = require('./MA4SJD6E.cjs');



var _O3ON5SWHcjs = require('./O3ON5SWH.cjs');


var _VPKU3AVXcjs = require('./VPKU3AVX.cjs');




var _M5IWBORBcjs = require('./M5IWBORB.cjs');


var _WT6AUJWVcjs = require('./WT6AUJWV.cjs');



var _3IVNQNIVcjs = require('./3IVNQNIV.cjs');

// src/menubar/menubar.tsx
var _jsxruntime = require('react/jsx-runtime');
var TagName = "div";
var useMenubar = _M5IWBORBcjs.createHook.call(void 0, 
  function useMenubar2(_a) {
    var _b = _a, {
      store: storeProp,
      composite = true,
      orientation: orientationProp,
      virtualFocus,
      focusLoop,
      rtl
    } = _b, props = _3IVNQNIVcjs.__objRest.call(void 0, _b, [
      "store",
      "composite",
      "orientation",
      "virtualFocus",
      "focusLoop",
      "rtl"
    ]);
    const context = _O3ON5SWHcjs.useMenubarProviderContext.call(void 0, );
    storeProp = storeProp || context;
    const store = _MA4SJD6Ecjs.useMenubarStore.call(void 0, {
      store: storeProp,
      orientation: orientationProp,
      virtualFocus,
      focusLoop,
      rtl
    });
    const orientation = store.useState(
      (state) => !composite || state.orientation === "both" ? void 0 : state.orientation
    );
    props = _WT6AUJWVcjs.useWrapElement.call(void 0, 
      props,
      (element) => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, _O3ON5SWHcjs.MenubarScopedContextProvider, { value: store, children: element }),
      [store]
    );
    if (composite) {
      props = _3IVNQNIVcjs.__spreadValues.call(void 0, {
        role: "menubar",
        "aria-orientation": orientation
      }, props);
    }
    props = _VPKU3AVXcjs.useComposite.call(void 0, _3IVNQNIVcjs.__spreadValues.call(void 0, { store, composite }, props));
    return props;
  }
);
var Menubar = _M5IWBORBcjs.forwardRef.call(void 0, function Menubar2(props) {
  const htmlProps = useMenubar(props);
  return _M5IWBORBcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useMenubar = useMenubar; exports.Menubar = Menubar;
