// @ariakit/react-core/cjs/__chunks/HHTQYHYL.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";



var _7K2RM2SDcjs = require('./7K2RM2SD.cjs');


var _M5IWBORBcjs = require('./M5IWBORB.cjs');

// src/dialog/dialog-context.tsx
var _react = require('react');
var ctx = _M5IWBORBcjs.createStoreContext.call(void 0, 
  [_7K2RM2SDcjs.DisclosureContextProvider],
  [_7K2RM2SDcjs.DisclosureScopedContextProvider]
);
var useDialogContext = ctx.useContext;
var useDialogScopedContext = ctx.useScopedContext;
var useDialogProviderContext = ctx.useProviderContext;
var DialogContextProvider = ctx.ContextProvider;
var DialogScopedContextProvider = ctx.ScopedContextProvider;
var DialogHeadingContext = _react.createContext.call(void 0, void 0);
var DialogDescriptionContext = _react.createContext.call(void 0, void 0);









exports.useDialogContext = useDialogContext; exports.useDialogScopedContext = useDialogScopedContext; exports.useDialogProviderContext = useDialogProviderContext; exports.DialogContextProvider = DialogContextProvider; exports.DialogScopedContextProvider = DialogScopedContextProvider; exports.DialogHeadingContext = DialogHeadingContext; exports.DialogDescriptionContext = DialogDescriptionContext;
