// @ariakit/react-core/cjs/__chunks/K32UV4VJ.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";


var _2P4N5BIRcjs = require('./2P4N5BIR.cjs');




var _M5IWBORBcjs = require('./M5IWBORB.cjs');



var _3IVNQNIVcjs = require('./3IVNQNIV.cjs');

// src/focus-trap/focus-trap.tsx
var TagName = "span";
var useFocusTrap = _M5IWBORBcjs.createHook.call(void 0, 
  function useFocusTrap2(props) {
    props = _3IVNQNIVcjs.__spreadProps.call(void 0, _3IVNQNIVcjs.__spreadValues.call(void 0, {
      "data-focus-trap": "",
      tabIndex: 0,
      "aria-hidden": true
    }, props), {
      style: _3IVNQNIVcjs.__spreadValues.call(void 0, {
        // Prevents unintended scroll jumps.
        position: "fixed",
        top: 0,
        left: 0
      }, props.style)
    });
    props = _2P4N5BIRcjs.useVisuallyHidden.call(void 0, props);
    return props;
  }
);
var FocusTrap = _M5IWBORBcjs.forwardRef.call(void 0, function FocusTrap2(props) {
  const htmlProps = useFocusTrap(props);
  return _M5IWBORBcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useFocusTrap = useFocusTrap; exports.FocusTrap = FocusTrap;
