// @ariakit/react-core/cjs/__chunks/N2JBBOXS.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";



var _3MQN464Lcjs = require('./3MQN464L.cjs');


var _M5IWBORBcjs = require('./M5IWBORB.cjs');

// src/tag/tag-context.tsx
var _react = require('react');
var TagValueContext = _react.createContext.call(void 0, null);
var TagRemoveIdContext = _react.createContext.call(void 0, 
  null
);
var ctx = _M5IWBORBcjs.createStoreContext.call(void 0, 
  [_3MQN464Lcjs.CompositeContextProvider],
  [_3MQN464Lcjs.CompositeScopedContextProvider]
);
var useTagContext = ctx.useContext;
var useTagScopedContext = ctx.useScopedContext;
var useTagProviderContext = ctx.useProviderContext;
var TagContextProvider = ctx.ContextProvider;
var TagScopedContextProvider = ctx.ScopedContextProvider;









exports.TagValueContext = TagValueContext; exports.TagRemoveIdContext = TagRemoveIdContext; exports.useTagContext = useTagContext; exports.useTagScopedContext = useTagScopedContext; exports.useTagProviderContext = useTagProviderContext; exports.TagContextProvider = TagContextProvider; exports.TagScopedContextProvider = TagScopedContextProvider;
