// @ariakit/react-core/cjs/__chunks/O3ON5SWH.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";



var _3MQN464Lcjs = require('./3MQN464L.cjs');


var _M5IWBORBcjs = require('./M5IWBORB.cjs');

// src/menubar/menubar-context.tsx
var _react = require('react');
var menubar = _M5IWBORBcjs.createStoreContext.call(void 0, 
  [_3MQN464Lcjs.CompositeContextProvider],
  [_3MQN464Lcjs.CompositeScopedContextProvider]
);
var useMenubarContext = menubar.useContext;
var useMenubarScopedContext = menubar.useScopedContext;
var useMenubarProviderContext = menubar.useProviderContext;
var MenubarContextProvider = menubar.ContextProvider;
var MenubarScopedContextProvider = menubar.ScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);








exports.useMenubarContext = useMenubarContext; exports.useMenubarScopedContext = useMenubarScopedContext; exports.useMenubarProviderContext = useMenubarProviderContext; exports.MenubarContextProvider = MenubarContextProvider; exports.MenubarScopedContextProvider = MenubarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
