// @ariakit/react-core/cjs/__chunks/RBHGB7EB.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";



var _3MQN464Lcjs = require('./3MQN464L.cjs');



var _AMARRBCFcjs = require('./AMARRBCF.cjs');


var _M5IWBORBcjs = require('./M5IWBORB.cjs');

// src/select/select-context.tsx
var _react = require('react');
var ctx = _M5IWBORBcjs.createStoreContext.call(void 0, 
  [_AMARRBCFcjs.PopoverContextProvider, _3MQN464Lcjs.CompositeContextProvider],
  [_AMARRBCFcjs.PopoverScopedContextProvider, _3MQN464Lcjs.CompositeScopedContextProvider]
);
var useSelectContext = ctx.useContext;
var useSelectScopedContext = ctx.useScopedContext;
var useSelectProviderContext = ctx.useProviderContext;
var SelectContextProvider = ctx.ContextProvider;
var SelectScopedContextProvider = ctx.ScopedContextProvider;
var SelectItemCheckedContext = _react.createContext.call(void 0, false);
var SelectHeadingContext = _react.createContext.call(void 0, null);









exports.useSelectContext = useSelectContext; exports.useSelectScopedContext = useSelectScopedContext; exports.useSelectProviderContext = useSelectProviderContext; exports.SelectContextProvider = SelectContextProvider; exports.SelectScopedContextProvider = SelectScopedContextProvider; exports.SelectItemCheckedContext = SelectItemCheckedContext; exports.SelectHeadingContext = SelectHeadingContext;
