// @ariakit/react-core/cjs/__chunks/TTVWSMCC.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";






var _O3ON5SWHcjs = require('./O3ON5SWH.cjs');



var _3MQN464Lcjs = require('./3MQN464L.cjs');



var _ZBR3CZBXcjs = require('./ZBR3CZBX.cjs');


var _M5IWBORBcjs = require('./M5IWBORB.cjs');

// src/menu/menu-context.tsx
var _react = require('react');
var menu = _M5IWBORBcjs.createStoreContext.call(void 0, 
  [_3MQN464Lcjs.CompositeContextProvider, _ZBR3CZBXcjs.HovercardContextProvider],
  [_3MQN464Lcjs.CompositeScopedContextProvider, _ZBR3CZBXcjs.HovercardScopedContextProvider]
);
var useMenuContext = menu.useContext;
var useMenuScopedContext = menu.useScopedContext;
var useMenuProviderContext = menu.useProviderContext;
var MenuContextProvider = menu.ContextProvider;
var MenuScopedContextProvider = menu.ScopedContextProvider;
var useMenuBarContext = _O3ON5SWHcjs.useMenubarContext;
var useMenuBarScopedContext = _O3ON5SWHcjs.useMenubarScopedContext;
var useMenuBarProviderContext = _O3ON5SWHcjs.useMenubarProviderContext;
var MenuBarContextProvider = _O3ON5SWHcjs.MenubarContextProvider;
var MenuBarScopedContextProvider = _O3ON5SWHcjs.MenubarScopedContextProvider;
var MenuItemCheckedContext = _react.createContext.call(void 0, 
  void 0
);













exports.useMenuContext = useMenuContext; exports.useMenuScopedContext = useMenuScopedContext; exports.useMenuProviderContext = useMenuProviderContext; exports.MenuContextProvider = MenuContextProvider; exports.MenuScopedContextProvider = MenuScopedContextProvider; exports.useMenuBarContext = useMenuBarContext; exports.useMenuBarScopedContext = useMenuBarScopedContext; exports.useMenuBarProviderContext = useMenuBarProviderContext; exports.MenuBarContextProvider = MenuBarContextProvider; exports.MenuBarScopedContextProvider = MenuBarScopedContextProvider; exports.MenuItemCheckedContext = MenuItemCheckedContext;
