// @ariakit/react-core/cjs/__chunks/2P4N5BIR.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";




var _M5IWBORBcjs = require('./M5IWBORB.cjs');



var _3IVNQNIVcjs = require('./3IVNQNIV.cjs');

// src/visually-hidden/visually-hidden.tsx
var TagName = "span";
var useVisuallyHidden = _M5IWBORBcjs.createHook.call(void 0, 
  function useVisuallyHidden2(props) {
    props = _3IVNQNIVcjs.__spreadProps.call(void 0, _3IVNQNIVcjs.__spreadValues.call(void 0, {}, props), {
      style: _3IVNQNIVcjs.__spreadValues.call(void 0, {
        border: 0,
        clip: "rect(0 0 0 0)",
        height: "1px",
        margin: "-1px",
        overflow: "hidden",
        padding: 0,
        position: "absolute",
        whiteSpace: "nowrap",
        width: "1px"
      }, props.style)
    });
    return props;
  }
);
var VisuallyHidden = _M5IWBORBcjs.forwardRef.call(void 0, function VisuallyHidden2(props) {
  const htmlProps = useVisuallyHidden(props);
  return _M5IWBORBcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useVisuallyHidden = useVisuallyHidden; exports.VisuallyHidden = VisuallyHidden;
