// @ariakit/react-core/cjs/__chunks/F4DU6MSG.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";



var _3MQN464Lcjs = require('./3MQN464L.cjs');



var _AMARRBCFcjs = require('./AMARRBCF.cjs');


var _M5IWBORBcjs = require('./M5IWBORB.cjs');

// src/combobox/combobox-context.tsx
var _react = require('react');
var ComboboxListRoleContext = _react.createContext.call(void 0, 
  void 0
);
var ctx = _M5IWBORBcjs.createStoreContext.call(void 0, 
  [_AMARRBCFcjs.PopoverContextProvider, _3MQN464Lcjs.CompositeContextProvider],
  [_AMARRBCFcjs.PopoverScopedContextProvider, _3MQN464Lcjs.CompositeScopedContextProvider]
);
var useComboboxContext = ctx.useContext;
var useComboboxScopedContext = ctx.useScopedContext;
var useComboboxProviderContext = ctx.useProviderContext;
var ComboboxContextProvider = ctx.ContextProvider;
var ComboboxScopedContextProvider = ctx.ScopedContextProvider;
var ComboboxItemValueContext = _react.createContext.call(void 0, 
  void 0
);
var ComboboxItemCheckedContext = _react.createContext.call(void 0, false);










exports.ComboboxListRoleContext = ComboboxListRoleContext; exports.useComboboxContext = useComboboxContext; exports.useComboboxScopedContext = useComboboxScopedContext; exports.useComboboxProviderContext = useComboboxProviderContext; exports.ComboboxContextProvider = ComboboxContextProvider; exports.ComboboxScopedContextProvider = ComboboxScopedContextProvider; exports.ComboboxItemValueContext = ComboboxItemValueContext; exports.ComboboxItemCheckedContext = ComboboxItemCheckedContext;
