// @ariakit/react-core/cjs/__chunks/P7J6L26K.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";


var _VQ52RGDHcjs = require('./VQ52RGDH.cjs');




var _M5IWBORBcjs = require('./M5IWBORB.cjs');



var _WT6AUJWVcjs = require('./WT6AUJWV.cjs');



var _3IVNQNIVcjs = require('./3IVNQNIV.cjs');

// src/button/button.tsx
var _dom = require('@ariakit/core/utils/dom');
var _react = require('react');
var TagName = "button";
var useButton = _M5IWBORBcjs.createHook.call(void 0, 
  function useButton2(props) {
    const ref = _react.useRef.call(void 0, null);
    const tagName = _WT6AUJWVcjs.useTagName.call(void 0, ref, TagName);
    const [isNativeButton, setIsNativeButton] = _react.useState.call(void 0, 
      () => !!tagName && _dom.isButton.call(void 0, { tagName, type: props.type })
    );
    _react.useEffect.call(void 0, () => {
      if (!ref.current) return;
      setIsNativeButton(_dom.isButton.call(void 0, ref.current));
    }, []);
    props = _3IVNQNIVcjs.__spreadProps.call(void 0, _3IVNQNIVcjs.__spreadValues.call(void 0, {
      role: !isNativeButton && tagName !== "a" ? "button" : void 0
    }, props), {
      ref: _WT6AUJWVcjs.useMergeRefs.call(void 0, ref, props.ref)
    });
    props = _VQ52RGDHcjs.useCommand.call(void 0, props);
    return props;
  }
);
var Button = _M5IWBORBcjs.forwardRef.call(void 0, function Button2(props) {
  const htmlProps = useButton(props);
  return _M5IWBORBcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.useButton = useButton; exports.Button = Button;
