// @ariakit/react-core/cjs/__chunks/WYGALZGN.cjs
"use strict";Object.defineProperty(exports, "__esModule", {value: true});"use client";


var _AH3CUB6Ncjs = require('./AH3CUB6N.cjs');


var _AMARRBCFcjs = require('./AMARRBCF.cjs');




var _M5IWBORBcjs = require('./M5IWBORB.cjs');



var _WT6AUJWVcjs = require('./WT6AUJWV.cjs');




var _3IVNQNIVcjs = require('./3IVNQNIV.cjs');

// src/popover/popover-arrow.tsx
var _dom = require('@ariakit/core/utils/dom');
var _misc = require('@ariakit/core/utils/misc');
var _react = require('react');
var _jsxruntime = require('react/jsx-runtime');
var TagName = "div";
var defaultSize = 30;
var halfDefaultSize = defaultSize / 2;
var rotateMap = {
  top: `rotate(180 ${halfDefaultSize} ${halfDefaultSize})`,
  right: `rotate(-90 ${halfDefaultSize} ${halfDefaultSize})`,
  bottom: `rotate(0 ${halfDefaultSize} ${halfDefaultSize})`,
  left: `rotate(90 ${halfDefaultSize} ${halfDefaultSize})`
};
function useComputedStyle(store) {
  const [style, setStyle] = _react.useState.call(void 0, );
  const contentElement = store.useState("contentElement");
  _WT6AUJWVcjs.useSafeLayoutEffect.call(void 0, () => {
    if (!contentElement) return;
    const win = _dom.getWindow.call(void 0, contentElement);
    const computedStyle = win.getComputedStyle(contentElement);
    setStyle(computedStyle);
  }, [contentElement]);
  return style;
}
var usePopoverArrow = _M5IWBORBcjs.createHook.call(void 0, 
  function usePopoverArrow2(_a) {
    var _b = _a, { store, size = defaultSize } = _b, props = _3IVNQNIVcjs.__objRest.call(void 0, _b, ["store", "size"]);
    const context = _AMARRBCFcjs.usePopoverContext.call(void 0, );
    store = store || context;
    _misc.invariant.call(void 0, 
      store,
      process.env.NODE_ENV !== "production" && "PopoverArrow must be wrapped in a Popover component."
    );
    const dir = store.useState(
      (state) => state.currentPlacement.split("-")[0]
    );
    const style = useComputedStyle(store);
    const fill = (style == null ? void 0 : style.getPropertyValue("background-color")) || "none";
    const stroke = (style == null ? void 0 : style.getPropertyValue(`border-${dir}-color`)) || "none";
    const borderWidth = (style == null ? void 0 : style.getPropertyValue(`border-${dir}-width`)) || "0px";
    const strokeWidth = Number.parseInt(borderWidth) * 2 * (defaultSize / size);
    const transform = rotateMap[dir];
    const children = _react.useMemo.call(void 0, 
      () => /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "svg", { display: "block", viewBox: "0 0 30 30", children: /* @__PURE__ */ _jsxruntime.jsxs.call(void 0, "g", { transform, children: [
        /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "path", { fill: "none", d: _AH3CUB6Ncjs.POPOVER_ARROW_PATH }),
        /* @__PURE__ */ _jsxruntime.jsx.call(void 0, "path", { stroke: "none", d: _AH3CUB6Ncjs.POPOVER_ARROW_PATH })
      ] }) }),
      [transform]
    );
    props = _3IVNQNIVcjs.__spreadProps.call(void 0, _3IVNQNIVcjs.__spreadValues.call(void 0, {
      children,
      "aria-hidden": true
    }, props), {
      ref: _WT6AUJWVcjs.useMergeRefs.call(void 0, store.setArrowElement, props.ref),
      style: _3IVNQNIVcjs.__spreadValues.call(void 0, {
        // server side rendering
        position: "absolute",
        fontSize: size,
        width: "1em",
        height: "1em",
        pointerEvents: "none",
        fill,
        stroke,
        strokeWidth
      }, props.style)
    });
    return _misc.removeUndefinedValues.call(void 0, props);
  }
);
var PopoverArrow = _M5IWBORBcjs.forwardRef.call(void 0, function PopoverArrow2(props) {
  const htmlProps = usePopoverArrow(props);
  return _M5IWBORBcjs.createElement.call(void 0, TagName, htmlProps);
});




exports.usePopoverArrow = usePopoverArrow; exports.PopoverArrow = PopoverArrow;
